Dockerfile

```diff
@@ -0,0 +1,34 @@
+# Use Python 3.10 slim image as base
+FROM python:3.10-slim
+
+# Set working directory
+WORKDIR /app
+
+# Install system dependencies
+RUN apt-get update && apt-get install -y \
+    gcc \
+    python3-dev \
+    curl \
+    && rm -rf /var/lib/apt/lists/*
+
+RUN curl -sSL https://install.python-poetry.org | python3 -
+ENV PATH="/root/.local/bin:$PATH"
```

Comment:
You can combine this with line 30 to remove this docker layer:


ENV PYTHONUNBUFFERED=1 \
    PATH="/root/.local/bin:$PATH"